realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/SyncSessionTests.java

```diff
@@ -455,4 +462,121 @@ public void run() {
         assertEquals(3, realm.where(StringOnly.class).count());
         realm.close();
     }
+
+    @Test
+    public void partialSync() throws InterruptedException {
+        SyncUser user = UserFactory.createUniqueUser(Constants.AUTH_URL);
+        SyncUser adminUser = UserFactory.createAdminUser(Constants.AUTH_URL);
+
+        final SyncConfiguration partialSyncConfig = configFactory
+                .createSyncConfigurationBuilder(user, Constants.SYNC_SERVER_URL)
+                .modules(new PartialSyncModule())
+                .withPartialSync()
+                .build();
+        SyncConfiguration adminConfig = configFactory
+                .createSyncConfigurationBuilder(adminUser, partialSyncConfig.getServerUrl().toString())
+                .modules(new PartialSyncModule())
+                .build();
+
+        // Using Admin user, populate the Realm.
+        Realm realm = Realm.getInstance(adminConfig);
+        realm.beginTransaction();
+        PartialSyncObjectA objectA = realm.createObject(PartialSyncObjectA.class);
```

Comment:
Create these in a loop instead? Would make it much more readable

Reply:
I'll refactor the creation of `PartialSyncObjectB` in a loop, `PartialSyncObjectA` string's has different value